### PR TITLE
Provide ability to run a pre bundle update script

### PR DESCRIPTION
**Pull request by joesustaric**
## Context
I was attempting to use this plugin, and came across an issue where I needed to install a app which was needed for the `bundle update` to pass.
The application that I'm configuring does not have it's 'own' docker container nor do I want to create and manage one seeing as thought the bundle update process will only be executed once a week or so. This means to me that build times are irrelevant and I don't mind the build taking a long time building a container.

This PR is the result of my 2nd approach, which seems _a lot_ simpler than my first approach.
This PR just uses a simple convention of 'put the script here and if it exists, it will run' similar to the `hooks`. I chose `.buildkite/scripts/pre-bundle-update` for this. `hooks` appears to be used more for the buildkite functionality so a scripts directory seemed more appropriate.

I've tested this very simply and it appears to work ok.

FYI the second approach was with a defined parameter in the pipeline.yml to point to the script directory. My noobness with writing buildkite plugins seem to get the better of me and I found it difficult to get it working correctly. (as you can see with the commit history)
So I defaulted back to the convention approach with also seems a lot simpler in implementation.

UPDATE

gone back to the script being passed in explicitly. Seems just as easy to just pass it into the docker command.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,11 +132,34 @@ the [Github Pull Request Buildkite Plugin] saves the PR number with the key
...
+2. You can specify a script location which will be executed prior to running the
+   bundle update. Here you can install and configure the container as needed.
+
+```yml
+steps:
+  - label: ":bundler: Update"
+    plugins:
+      - envato/bundle-update#v0.7.0:
+          update: true
+          pre-bundle-update: .buildkite/scripts/pre-bundle-update
```
Comment (orien) [inline README.md]: We should add this parameter to the [configuration section](https://github.com/envato/bundle-update-buildkite-plugin#configuration) of the readme too.

File: update/update.sh [hunk 2]
```diff
@@ -2,4 +2,12 @@
 set -euo pipefail
 
 cd /bundle_update
+
+if [ -f "$PRE_BUNDLE_UPDATE" ]; then
```
Comment (orien) [inline update/update.sh]: Perhaps we don't need this conditional. If the script doesn't exist, I'm thinking it makes sense that the bundle update should fail.

Comment (orien) [reply update/update.sh]: Additionally, as we're using `eval` on [line 8](https://github.com/envato/bundle-update-buildkite-plugin/pull/7/files#diff-badfd8b06784e6c545bd35d65105e7ecR8), it doesn't have to be a script. A straight up command could be provided:

```yml
steps:
  - label: ":bundler: Update"
    plugins:
      - envato/bundle-update#v0.7.0:
          update: true
          pre-bundle-update: "apk add --no-progress build-base"
```

Comment (joesustaric) [reply update/update.sh]: Interesting. Yep you're right it should fail if you have spedified a script and it does not exist. I will update that.

In regards to providing commands via the yml.. I'm not a fan of that tbh. 
It feels nicer if a script holds all these commands, easier to locate run, debug, etc.. 
I might remove the `eval` and just do a `.`

Comment (orien) [reply update/update.sh]: It does provide options to others who may think differently though:

> A whole extra file to store one command

And it's consistent with the top level Buildkite step `command` attribute.

Comment (joesustaric) [reply update/update.sh]: humm ok fair enough.. I'll trust that developers wont write a 300 line shell script as a yaml parameter :D

**Pull request merged**